.github/workflows/appsignals-e2e-ec2-test.yml

```diff
@@ -0,0 +1,191 @@
+# This is a reusable workflow for running the E2E test for App Signals.
+# It is meant to be called from another workflow.
+# Read more about reusable workflows: https://docs.github.com/en/actions/using-workflows/reusing-workflows#overview
+name: App Signals Enablement E2E Testing - EC2 Use Case
+on:
+  workflow_call:
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  AWS_DEFAULT_REGION: us-east-1
+  APP_SIGNALS_E2E_TEST_ACCOUNT_ID: ${{ secrets.APP_SIGNALS_E2E_TEST_ACCOUNT_ID }}
+  SAMPLE_APP_FRONTEND_SERVICE_JAR: "s3://aws-appsignals-sample-app/main-service.jar"
+  SAMPLE_APP_REMOTE_SERVICE_JAR: "s3://aws-appsignals-sample-app/remote-service.jar"
+  APP_SIGNALS_ADOT_JAR: "https://github.com/aws-observability/aws-otel-java-instrumentation/releases/latest/download/aws-opentelemetry-agent.jar"
+  APP_SIGNALS_CW_AGENT_RPM: s3://${{ secrets.S3_INTEGRATION_BUCKET }}/integration-test/binary/${{ github.sha }}/amazon_linux/amd64/latest/amazon-cloudwatch-agent.rpm
+  METRIC_NAMESPACE: AppSignals
+  LOG_GROUP_NAME: /aws/appsignals/generic
+
+jobs:
+  e2e-ec2-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get testing resources from ADOT
+        uses: actions/checkout@v4
+        with:
+          repository: aws-observability/aws-otel-java-instrumentation
+          ref: main
+          
+      - name: Download patch action script
+        uses: actions/checkout@v4
+        with:
+          path: patch-dependencies
+          sparse-checkout: |
+            .github/actions/patch-dependencies/action.yml
+
+      - name: Replace patch dependency action.yml
+        run: |
+          cp -f ./patch-dependencies/.github/actions/patch-dependencies/action.yml ./.github/actions/patch-dependencies/action.yml
+
+      - uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: temurin
+
+      - name: Generate testing id
+        run: echo TESTING_ID="${{ github.run_id }}-${{ github.run_number }}" >> $GITHUB_ENV
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.APP_SIGNALS_E2E_IAM_ROLE }}
+          aws-region: ${{ env.AWS_DEFAULT_REGION }}
+
+      - name: Update main.tf to download the new cw-agent.rpm
+        working-directory: testing/terraform/ec2
+        run: |
+          old_command='"wget -O cw-agent.rpm ${var.cw_agent_rpm}",'
+          new_command='"aws s3 cp ${{ env.APP_SIGNALS_CW_AGENT_RPM }} ./cw-agent.rpm",'
+          sed -i "s|$old_command|$new_command|" "main.tf"
```

Comment:
This is not a great way to do this, `sed` commands often silently fail but I see why we're doing this given we're changing from wget to aws cli. I'd think twice about it though, maybe we can just update the terraform itself to deal with this somehow.

Reply:
I did consider that option, but that will require us to make change on the OTEL repo. I added a validation step for now to check if the sed command worked correctly.